site/docs/hosting/supabase.md

```diff
@@ -0,0 +1,95 @@
+
+This guide tells you about the ways you can host your grammY bots on [Supabase](https://supabase.com/).
+
+Please note that you need to have a [GitHub](https://github.com) account before you can use [Supabase Edge Functions](https://supabase.com/docs/guides/functions).
+Moreover, Supabase Edge Functions are based on [Deno Deploy](https://deno.com/deploy), so just like [our Deno Deploy guide](./deno-deploy.md), this guide is only for Deno users of grammY.
+
+Supabase Edge Functions is ideal for most simple bots, and you should note that not all Deno features are available for apps running on Supabase Edge Functions.
+For example, there is no file system on Supabase Edge Functions.
+It's just like the other many serverless platforms, but dedicated for Deno apps.
+
+The result of this tutorial [can be seen in our example bots repository](https://github.com/grammyjs/examples/tree/main/Supabase-edge-functions).
+
+## Setup
+
+In order to deploy a Supabase Edge Function, you will need to create a Supabase account, install their CLI, and create a Supabase project.
+You should first [follow their documentation](https://supabase.com/docs/guides/functions#prerequisites) to set things up.
+
+Create a new supabase function by running this command.
+
+```sh
+supabase functions new telegram-bot
+```
+
+Once you have created a project with your Supabase Edge function, you can write your bot.
+
+## Preparing Your Code
+
+> Remember that you need to [run your bot on webhhoks](../guide/deployment-types.md#how-to-use-webhooks), so you should use `webhookCallback` and not call `bot.start()` in your code.
+
+You can use this short example bot as a starting point.
+
+```ts
+import { serve } from "https://deno.land/std@0.140.0/http/server.ts";
+import { Bot, webhookCallback } from "https://deno.land/x/grammy/mod.ts";
+
+const bot = new Bot(Deno.env.get("BOT_TOKEN") ?? "");
+
+bot.command("start", (ctx) => ctx.reply("Welcome! Up and running."));
+bot.command("ping", (ctx) => ctx.reply(`Pong! ${new Date()}`));
+
+const handleUpdate = webhookCallback(bot, "std/http");
+
+serve(async (req) => {
+  try {
+    const url = new URL(req.url);
+    if (url.searchParams.get("secret") !== bot.token) {
+      return new Response("not allowed", { status: 405 });
+    }
+    return await handleUpdate(req);
+  } catch (err) {
+    console.error(err);
+  }
+});
+```
+
+## Deploying
+
+You can now deploy your bot to Supabase
+Note that you will have to disable JWT authorization because Telegram uses a different way to make sure the requests are coming from Telegram.
+You can deploy the function using this command.
+
+```sh
+supabase functions deploy --no-verify-jwt telegram-bot
+```
+
+Next, you need to give your bot token to Supabase so that your bot has access to it as an environment variable.
+
+```sh
+# Replace 123:aBcDeF-gh with your real bot token.
+supabase secrets set BOT_TOKEN=123:aBcDeF-gh
+```
+
+Your Supabase Edge Function is now working.
```

Comment:
```suggestion
Your Supabase Function is now working.
```

Reply:
Then it would also have to be Supabase Edge Function project above.

Reply:
Forget it, I read the diff the other way around. I will apply it.